tests/pipeline/PipelineCTracing.cpp

```diff
@@ -0,0 +1,109 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/PipelineC/PipelineC.h"
+#include "revng/PipelineC/Tracing/Private.h"
+#include "revng/PipelineC/Tracing/Trace.h"
+#include "revng/Support/TemporaryFile.h"
+
+#define BOOST_TEST_MODULE PipelineCTracing
+bool init_unit_test();
+#include "boost/test/unit_test.hpp"
+
+#include "revng/UnitTestHelpers/UnitTestHelpers.h"
+
+namespace rt = revng::tracing;
+
+class TheHarness {
+public:
+  std::string Buffer;
+  std::optional<rt::Trace> Trace;
+
+public:
+  bool parseTrace() {
+    auto MemoryBuffer = llvm::MemoryBuffer::getMemBuffer(Buffer);
+    auto MaybeTrace = rt::Trace::fromBuffer(*MemoryBuffer);
+    if (!MaybeTrace) {
+      llvm::consumeError(MaybeTrace.takeError());
+      return false;
+    }
+    Trace = MaybeTrace.get();
+    return true;
+  }
+};
+
+static TheHarness *Harness;
+
+struct Fixture {
+public:
+  Fixture() {
+    using ArgvT = std::vector<const char *>;
+    ArgvT Argv = { "",
+                   "-load=lib/librevngStringContainerLibrary.so",
+                   "-pipeline-path=tests/pipeline/Tracing.yml" };
```

Comment:
You're assuming the current directory, consider using `llvm::sys::fs::getMainExecutable`.

Reply:
Switched

Reply:
I still see relative paths